klippa/src/main.rs

```diff
@@ -52,15 +31,15 @@ struct Args {
 fn main() {
     let args = Args::parse();
 
-    let gids = match populate_gids(&args.gids) {
+    let gids = match populate_gids(&args.gids.unwrap_or(String::new())) {
```

Comment:
- as written, you can do `unwrap_or_default()`.
- it seems a bit funny to call this fn with an empty string when there's no argument, so I might prefer something like,

```rust
let gids = args.gids.as_ref().map(populate_gids).transpose()?.unwrap_or_default();
```
this is imagining that you're in a function that returns a result, and so can use `?`. (`.transpose()` is a method that turns `Option<Result<T, E>>` into `Result<Option<T>, E>`)

Also I would suggest a minor refactor here: instead of having a bunch of different places where you catch an error, you can have a single `fn run() -> Result<(), SubsetError>` and in your `main()` all you do is call that function and print any errors; that function would have all of your actual logic.

Reply:
I changed to `unwrap_or_default()`. And I'll refactor the code as you suggested later in another PR.
I tried your preferred format but I got an error because `args.gids.as_ref() `is an `Option<&str>`, while the input of `populate_gids` is `&str`.